### PR TITLE
Finish the integer purge

### LINKED ISSUE #688: docs: remove reference to Oscar.Integer, Oscar.Rational; update or remove `IntegerTypes.svg` and `RationalTypes.svg`
Some time ago, `Oscar.Integer` was removed, but the Oscar manual still refers to it in `docs/src/Rings/integer.md` and `docs/src/Rings/rational.md` as well as in `docs/src/img/IntegerTypes.svg`.

Likewise, `Oscar.Rational` is currently identical to `Base.Rational`, but `docs/src/Rings/rational.md` and `docs/src/img/RationalTypes.svg` instead claim that it is a type that represents "Oscar rational numbers" resp. `fmpq`.

This should be corrected.

### PR DESCRIPTION
**Pull request by tthsqe12**
probably closes #688 
there is also now https://github.com/Nemocas/Nemo.jl/issues/1178

### CODE REVIEW

File: docs/src/Rings/integer.md [hunk 1]
```diff
@@ -18,27 +18,23 @@ by default. This means that integers typed at the
...
-The `Base.BigInt` type is backed by GMP multiprecision integers and can
-represent integers whose size is usually only limited by available memory.
-
-Oscar currently only has one integer type, `Oscar.fmpz` which for performance
+relevant here are `Int` and `BigInt`. All the Julia integer types
+belong to `Integer`.
+
+The `Int` type is for machine integers which are highly efficient, but
+can only represent integers up to a certain size, and most basic arithmetic
+operations are performed unchecked, that is, they can silently overflow. The
```
Comment (fingolfin) [inline docs/src/Rings/integer.md]: Perhaps we should insert the example `2^100 == 0` somewhere around here?

File: docs/src/Rings/integer.md [hunk 2]
```diff
@@ -246,12 +239,9 @@ with ``|r| < |b|``.
 
 ### Division with remainder
 
-* `divrem(a::Oscar.Integer, b::Oscar.Integer) -> (Oscar.Integer, Oscar.Integer)` : division with remainder
-* `div(a::Oscar.Integer, b::Oscar.Integer) -> Oscar.Integer` : quotient only
-* `rem(a::Oscar.Integer, b::Oscar.Integer) -> Oscar.Integer` : remainder only
-
-One or both arguments may be Julia integers, however if they both are, the
-result will be a Julia integer not an Oscar integer.
```
Comment (fingolfin) [inline docs/src/Rings/integer.md:254]: Technically speaking, this comment is still valid (I just tried), except that now I guess "Oscar integer" should be `fmpz`. Are you intentionally removing it?

File: docs/src/Rings/integer.md [hunk 3]
```diff
@@ -273,10 +263,7 @@ div(ZZ(2), ZZ(0))
 
 ### Modular reduction
 
-* `mod(a::Oscar.Integer, b::Oscar.Integer) -> Oscar.Integer` : remainder only
-
-One or both arguments may be Julia integers, however if they both are, the
-result will be a Julia integer not an Oscar integer.
```
Comment (fingolfin) [inline docs/src/Rings/integer.md:279]: ditto

**Review by fingolfin — APPROVED** [review]
Thanks for working on this. Some minor remarks/questions, but overall looks fine to me. Perhaps @fieker or @thofma have anything to add? Or perhaps they just want to merge it?

File: docs/src/Rings/integer.md (hunk 2 above)
Comment (tthsqe12) [reply docs/src/Rings/integer.md:254]: Yes, I am removing it. Its seems rather pedantic to point out that we haven't changed `divrem(Int, Int)->(Int, Int)`. and for gcd, gcdx, lcm, ...

File: docs/src/Rings/integer.md (hunk 1 above)
Comment (tthsqe12) [reply docs/src/Rings/integer.md]: 2^64 -> 0 appears in the powering section below. I inserted a link to julia's overflow section.

File: docs/src/Rings/integer.md (hunk 2 above)
Comment (fingolfin) [reply docs/src/Rings/integer.md:254]: Sure, that bit could be removed. But this also removes the mention that `divrem(Int,fmpz)` is supported and yields an `fmpz`?

Comment (tthsqe12) [reply docs/src/Rings/integer.md:254]: I put a general remark about this in the "Julia integers in Oscar functions" section.

**Pull request merged**